### PR TITLE
ENH add 3x with correct run exports

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This PR adds a build of 3x with run exports up to 5. The goal is to have a version that prevents excessive repodata patching in https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/issues/250.

We should merge this to a 3x branch in the upstream feedstock. I'm happy to maintain that.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by minrk** [comment]
added 3.x branch, if you want to retarget the PR. Probably want to add 3.x to [`bot.abi_migration_branches`](https://conda-forge.org/docs/maintainer/conda_forge_yml.html#bot) in conda-forge.yml

File: recipe/meta.yaml [hunk 1]
```diff
@@ -37,7 +37,7 @@ outputs:
     script: build-mpi.sh
     build:
       run_exports:
-        - {{ pin_subpackage('mpich', max_pin='x') }}
+        - mpich >={{ version }},<5.0a0
```
Comment (dalcinl) [inline recipe/meta.yaml:40]: MPICH has a commitment to ABI compatibility. v4.0 is supposed to be ABI compatible with any other v4.x build.

**Comment by dalcinl** [comment]
Should't you use a branch to track changes for MPICH v3.x releases?

**Comment by beckermr** [comment]
@minrk @conda-forge/mpich this is ready for review and merge!

**Pull request merged**